storm-core/src/jvm/backtype/storm/StormSubmitter.java

```diff
@@ -442,4 +444,36 @@ public static String submitJar(Map conf, String localJar, ProgressListener liste
          */
         public void onCompleted(String srcFile, String targetFile, long totalBytes);
     }
+    
+
+    private static void validateConfs(Map stormConf, StormTopology topology) throws IllegalArgumentException {
+        LOG.info("Validating storm Confs...");
+        double largestMemReq = getMaxExecutorMemoryUsageForTopo(topology, stormConf);
+        Double topologyWorkerMaxHeapSize = Utils.getDouble(stormConf.get(Config.TOPOLOGY_WORKER_MAX_HEAP_SIZE_MB));
+        if(topologyWorkerMaxHeapSize < largestMemReq) {
+            throw new IllegalArgumentException("Topology will not be able to be successfully scheduled: Config TOPOLOGY_WORKER_MAX_HEAP_SIZE_MB="
+                    +Utils.getDouble(stormConf.get(Config.TOPOLOGY_WORKER_MAX_HEAP_SIZE_MB)) + " < " 
+                            + largestMemReq + " (Largest memory requirement of a component in the topology). Perhaps set TOPOLOGY_WORKER_MAX_HEAP_SIZE_MB to a larger amount");
+        }
+    }
+
+
+    private static double getMaxExecutorMemoryUsageForTopo(StormTopology topology, Map topologyConf) {
+        double largestMemoryOperator = 0.0;
+        for(Map<String, Double> entry : backtype.storm.scheduler.resource.Utils.getBoltsResources(topology, topologyConf).values()) {
```

Comment:
Can we rename backtype.storm.scheduler.resource.Utils to be something like ResourceUtils instead.  Just so there is no conflict with imports.  I think it will make the code cleaner.